### ISSUE TITLE
Set fixed values like the emulator in virt-handler instead of virt-controller

### ISSUE THREAD

**Issue by rmohr** [kubevirt/kubevirt#124]
Currently we set some Domain attributes like UUID, Name and Emulator in virt-controller.

They should actually be set in the domain manager in virt-handler and these fields should not even be part of v1.DomainSpec. Emulator is not important for the cluster wide view and UUID and Name are duplicates of the v1.VM metadata.

 - [ ] Remove these fields from v1.DomainSpec
 - [ ] Set fields in manager.SyncVM before starting or updating a VM

**Comment by rmohr** [comment]
Additional fields which do not really make sense to expose:

```yaml
os:
  type:
    os: hvm
type: qemu
```

We only support qemu and full virtualization (at the moment).

**Comment by rmohr** [comment]
As @mpolednik suggested some time ago, we can probably also move the `spec.domain` subsection up to `spec` and merge it with the fields there.

**Comment by fabiand** [comment]
Mh - I do like that we keep the domain part separate form the complete VM spec.
It might become valuable if we start thinking about HA. i.e.

```
spec:
   ha-stand-by-instances-foo: 2
   domain:
      ...
```

thus - I can currently imagine that we get to a state where we need to keep non-domain specific, but cluster wide infos and then it makes sense to have the domain specification in a subsection.

@mpolednik ?

**Comment by fabiand** [comment]
Note on the original comment: In oVirt we know that some users want to set the specific emulator - there it's solved by using a hook.
